src/me/mrCookieSlime/Slimefun/SlimefunPlugin.java

```diff
@@ -268,49 +264,7 @@ else if (getDescription().getVersion().startsWith("RC - ")) {
 
 			// Armor Update Task
 			if (config.getBoolean("options.enable-armor-effects")) {
-				getServer().getScheduler().runTaskTimer(this, () -> {
-					for (Player p: Bukkit.getOnlinePlayers()) {
-						for (ItemStack armor: p.getInventory().getArmorContents()) {
-							if (armor != null && Slimefun.hasUnlocked(p, armor, true)) {
-								if (SlimefunItem.getByItem(armor) instanceof SlimefunArmorPiece) {
-									for (PotionEffect effect: ((SlimefunArmorPiece) SlimefunItem.getByItem(armor)).getEffects()) {
-										p.removePotionEffect(effect.getType());
-										p.addPotionEffect(effect);
-									}
-								}
-								if (SlimefunManager.isItemSimiliar(armor, SlimefunItem.getItem("SOLAR_HELMET"), false) && p.getWorld().getTime() < 12300 || p.getWorld().getTime() > 23850 && p.getEyeLocation().getBlock().getLightFromSky() == 15) {
-									ItemEnergy.chargeInventory(p, Float.valueOf(String.valueOf(Slimefun.getItemValue("SOLAR_HELMET", "charge-amount"))));
-								}
-							}
-						}
-
-						for (ItemStack radioactive: utilities.radioactiveItems) {
-							if (p.getInventory().containsAtLeast(radioactive, 1) || SlimefunManager.isItemSimiliar(p.getInventory().getItemInOffHand(), radioactive, true)) {
-								// Check if player is wearing the hazmat suit
-								// If so, break the loop
-								if (SlimefunManager.isItemSimiliar(SlimefunItems.SCUBA_HELMET, p.getInventory().getHelmet(), true) &&
-										SlimefunManager.isItemSimiliar(SlimefunItems.HAZMATSUIT_CHESTPLATE, p.getInventory().getChestplate(), true) &&
-										SlimefunManager.isItemSimiliar(SlimefunItems.HAZMATSUIT_LEGGINGS, p.getInventory().getLeggings(), true) &&
-										SlimefunManager.isItemSimiliar(SlimefunItems.RUBBER_BOOTS, p.getInventory().getBoots(), true)) {
-									break;
-								}
-
-								// If the item is enabled in the world, then make radioactivity do its job
-								if (Slimefun.isEnabled(p, radioactive, false)) {
-									p.addPotionEffect(new PotionEffect(PotionEffectType.WITHER, 400, 3));
-									p.addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, 400, 3));
-									p.addPotionEffect(new PotionEffect(PotionEffectType.CONFUSION, 400, 3));
-									p.addPotionEffect(new PotionEffect(PotionEffectType.WEAKNESS, 400, 3));
-									p.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, 400, 1));
-									p.addPotionEffect(new PotionEffect(PotionEffectType.SLOW_DIGGING, 400, 1));
-									p.setFireTicks(400);
-									break; 
-									// Break the loop to save some calculations
-								}
-							}
-						}
-					}
-				}, 0L, config.getInt("options.armor-update-interval") * 20L);
+				getServer().getScheduler().runTaskTimer(this, new ArmorTask(), 0L, config.getInt("options.armor-update-interval") * 20L);
```

Comment:
Do we need this to be sync? The only thing I can think which MAY need to be sync is the potion effects. I haven't ever tried to add them async so I don't know if they're thread-safe or not. We should definitely make it async though even if we just offload the potion effects to a sync thread.

Reply:
I mean I don't think Bukkit.getOnlinePlayers() can even be sync, that would very likely cause lots of ConcurrentModificationExceptions, wouldn't it?
Or is getOnlinePlayers() actually thread-safe?

Reply:
There are no modifications and it's immutable so thread-safe.

Reply:
It's not modified when players join and leave?
Is it just an immutable copy of the current state?

Reply:
IIRC it's just an immutable list of the current net clients player

Reply:
Alright then, sync it shall be!